addons/http_routing/models/ir_http.py

```diff
@@ -415,7 +415,11 @@ def _match(cls, path):
         else:
             url_lang_str = ''
             path_no_lang = path
-        allow_redirect = request.httprequest.method != 'POST'
+
+        allow_redirect = (
+            request.httprequest.method != 'POST'
+            and getattr(request, 'is_frontend_multilang', True)
```

Comment:
In case we don't match a backend route, nor `is_frontend` nor `is_frontend_multilang` are set yet, they'll be after the second `super()._match()` at the end of the method.

In order for `and request.is_frontend_multilang` to work, I would need to apply the following diff too:

```diff
diff --git a/addons/http_routing/models/ir_http.py b/addons/http_routing/models/ir_http.py
index 3579a10f93bc..7dbac279fd58 100644
--- a/addons/http_routing/models/ir_http.py
+++ b/addons/http_routing/models/ir_http.py
@@ -412,6 +412,11 @@ class IrHttp(models.AbstractModel):
         # See /1, match a non website endpoint
         try:
             rule, args = super()._match(path)
             routing = rule.endpoint.routing
             request.is_frontend = routing.get('website', False)
             request.is_frontend_multilang = request.is_frontend and routing.get('multilang', routing['type'] == 'http')
             if not request.is_frontend:
                 return rule, args
         except NotFound:
             _, url_lang_str, *rest = path.split('/', 2)
             path_no_lang = '/' + (rest[0] if rest else '')
+            # Assume we didn't match any controller because there is a
+            # lang prefix present in the URL and that ultimately we will
+            # match a route once that lang prefix is removed. The two
+            # attributes are set again after the second super().match().
+            request.is_frontend = True
+            request.is_frontend_multilang = True
         else:
             url_lang_str = ''
             path_no_lang = path
```

I prefer the `getattr()` way, what do you thing?

Reply:
If you go to `/fr/web/health`, it doesn't match any route thus would set `request.is_frontend=True`, it processes to remove the lang prefix and call `super().match()` again this time with `/web/health`, this time we match a controller and we set `request.is_frontend=False` as the route is not website.

Using the `getattr()`-way, we never set a wrong `request.is_frontend=True` just for the sake of making the code `... and request.is_frontend_multilang` work.

Reply:
The catch is that going to `/fr/web/health` was possible even pre-httpocalypse, it just set the frontend_lang cookies and proceed to the controller. At some point during my reflexion I challenged the idea to raise an exception or to redirect the browser on such URLs. Ultimately I decided to keep it to `[REF]` and to not change that behavior.

Reply:
> I prefer the getattr() way, what do you thing?

Sure, it's fine then. I wasn't implying to have new diff to make it work, I thought you wrote it like that and that you forgotit could be working without the `getattr()`, as I remembered multiple discussion where the `getattr()` usage were "deprecated" :+1: